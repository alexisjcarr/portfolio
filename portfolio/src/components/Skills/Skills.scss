.container {
  display: flex;
  flex-direction: column;
  .skills-card {
    background-color: yellowgreen;
    width: 90%;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    color: black;

    h1 {
      @media (max-width: 800px) {
        font-size: 20px;
      }
    }

    h3 {
      text-decoration: underline;
      @media (max-width: 800px) {
        font-size: 15px;
      }
    }

    .skills {
      display: flex;
      flex-wrap: wrap;
      justify-content: center;

      @media (max-width: 800px) {
        justify-content: center;
      }

      .skill {
        padding: 20px;
        margin: 20px;

        p {
          @media (max-width: 800px) {
            font-size: 10px;
          }
        }

        .icon {
          font-size: 150px;

          @media (max-width: 800px) {
            font-size: 80px;
          }
        }

        @media (max-width: 800px) {
          padding: 0px;
          margin: 5px;
        }
      }
    }
  }
}
