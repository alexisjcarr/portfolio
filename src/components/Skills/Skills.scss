.container {
  .skills-card {
    background-color: yellowgreen;
    width: 90%;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    color: black;

    h1 {
      @media (max-width: 500px) {
        font-size: 20px;
      }
    }

    .skills {
      display: flex;
      flex-wrap: wrap;
      justify-content: space-between;

      @media (max-width: 500px) {
        justify-content: none;
      }

      .skill {
        padding: 20px;
        margin: 20px;

        p {
          @media (max-width: 500px) {
            font-size: 10px;
          }
        }

        .icon {
          font-size: 150px;

          @media (max-width: 500px) {
            font-size: 80px;
          }
        }

        @media (max-width: 500px) {
          padding: 0px;
          margin: 5px;
        }
      }
    }
  }
}