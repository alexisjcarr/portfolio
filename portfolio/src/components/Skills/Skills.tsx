import * as React from 'react'

import './Skills.scss'

const Skills: React.SFC = () => {
  return (
    <div className="container" id="skills">
      <div className="skills-card">
        <h1>skills</h1>
        <hr />
        <h3>Industry Knowledge</h3>
        <div className="skills">
          <div className="skill">
            <i className="icon fab fa-aws"></i>
            <p>AWS</p>
            <p>(Certified Solutions</p>
              <p>Architect)</p>
          </div>
          <div className="skill">
            <i className="icon fas fa-project-diagram"></i>
            <p>Systems Design</p>
          </div>
          <div className="skill">
            <i className="icon fas fa-cogs"></i>
            <p>Microservices</p>
          </div>
          <div className="skill">
            <i className="icon fab fa-git"></i>
            <p>git</p>
          </div>
          <div className="skill">
            <i className="icon fas fa-flask"></i>
            <p>Testing</p>
          </div>
        </div>
      </div>
      
      <div className="skills-card">
        <hr />
        <h3>Languages</h3>
        <div className="skills">
          <div className="skill">
              <i className="icon fab fa-python"></i>
              <p>Python</p>
            </div>
            <div className="skill">
              <i className="icon fab fa-js-square"></i>
              <p>JavaScript/</p>
              <p>TypeScript</p>
            </div>
            <div className="skill">
              <i className="icon fas fa-registered"></i>
              <p>R</p>
            </div>
            <div className="skill">
              <i className="icon fab fa-java"></i>
              <p>Java/Kotlin</p>
            </div>
            <div className="skill">
              <i className="icon fab fa-google"></i>
              <p>Golang</p>
            </div>
            <div className="skill">
              <i className="icon fas fa-window-maximize"></i>
              <p>Bash</p>
            </div>
          </div>
      </div>

      <div className="skills-card">
        <hr />
        <h3>Platforms</h3>
        <div className="skills">
          <div className="skill">
            <i className="icon fas fa-network-wired"></i>
            <p>Kubernetes</p>
          </div>
          <div className="skill">
            <i className="icon fab fa-docker"></i>
            <p>Docker</p>
          </div>
          <div className="skill">
            <i className="icon fab fa-node"></i>
            <p>Node/</p>
            <p>Express</p>
          </div>
          <div className="skill">
            <i className="icon fas fa-rocket"></i>
            <p>Micronaut</p>
          </div>
          <div className="skill">
            <i className="icon fas fa-seedling"></i>
            <p>Spring Boot</p>
          </div>
          <div className="skill">
            <i className="icon far fa-window-maximize"></i>
            <p>MATLAB</p>
          </div>
          </div>
      </div>

      <div className="skills-card">
        <hr />
        <h3>Frameworks</h3>
        <div className="skills">
          <div className="skill">
          <i className ="icon fas fa-cog"></i>
          <p>gRPC</p>
          </div>
          <div className="skill">
          <i className="icon fab fa-react"></i>
          <p>React/Redux</p>
          </div>
        </div>
      </div>

      <div className="skills-card">
        <hr />
        <h3>Databases</h3>
        <div className="skills">
          <div className="skill">
            <i className="icon fas fa-database"></i>
            <p>MySQL</p>
          </div>
          <div className="skill">
            <i className="icon fas fa-server"></i>
            <p>DynamoDB</p>
          </div>
          <div className="skill">
            <i className="icon fab fa-searchengin"></i>
            <p>Elasticsearch</p>
          </div>
      </div>
      </div>
    </div>
  )
}

export default Skills
