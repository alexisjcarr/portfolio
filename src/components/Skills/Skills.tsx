import * as React from "react";

import "./Skills.scss";

export interface SkillsProps {}

const Skills: React.SFC<SkillsProps> = () => {
  return (
    <div className="container" id="skills">
      <div className="skills-card">
        <h1>skills</h1>
        <div className="skills">
          <div className="skill">
            <i className="icon fab fa-react"></i>
            <p>React</p>
          </div>
          <div className="skill">
            <i className="icon fab fa-node"></i>
            <p>Node/Express</p>
          </div>
          <div className="skill">
            <i className="icon fab fa-html5"></i>
            <p>HTML</p>
          </div>
          <div className="skill">
            <i className="icon fab fa-css3-alt"></i>
            <p>CSS</p>
          </div>
          <div className="skill">
            <i className="icon fab fa-js-square"></i>
            <p>Javascript</p>
          </div>
          <div className="skill">
            <i className="icon fab fa-sass"></i>
            <p>Sass</p>
          </div>
          <div className="skill">
            <i className="icon fab fa-python"></i>
            <p>Python</p>
          </div>
          <div className="skill">
            <i className="icon fab fa-git"></i>
            <p>git</p>
          </div>
          <div className="skill">
            <i className="icon fas fa-database"></i>
            <p>Databases</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Skills;
