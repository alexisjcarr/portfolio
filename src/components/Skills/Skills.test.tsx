import React from "react";
import { render, cleanup } from "@testing-library/react";

import Skills from "./Skills";

describe("<Skills />", () => {
  let originalWarning: {
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
    },
    originalError: {
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
    };

  beforeEach(() => {
    originalWarning = console.warn;
    originalError = console.error;
    console.error = jest.fn();
    console.warn = jest.fn(); // temporarily sets console.warn to a jest function
  });

  afterEach(() => {
    cleanup();

    console.warn = originalWarning; // turns it back into the original error after test runs
    console.error = originalError;
  });

  it("matches snapshot", () => {
    expect(render(<Skills />)).toMatchSnapshot();
  });
});
