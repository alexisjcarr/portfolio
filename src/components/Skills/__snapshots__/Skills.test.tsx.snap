// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skills /> matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        id="skills"
      >
        <div
          class="skills-card"
        >
          <h1>
            skills
          </h1>
          <div
            class="skills"
          >
            <div
              class="skill"
            >
              <i
                class="icon fab fa-react"
              />
              <p>
                React
              </p>
            </div>
            <div
              class="skill"
            >
              <i
                class="icon fab fa-node"
              />
              <p>
                Node/Express
              </p>
            </div>
            <div
              class="skill"
            >
              <i
                class="icon fab fa-html5"
              />
              <p>
                HTML
              </p>
            </div>
            <div
              class="skill"
            >
              <i
                class="icon fab fa-css3-alt"
              />
              <p>
                CSS
              </p>
            </div>
            <div
              class="skill"
            >
              <i
                class="icon fab fa-js-square"
              />
              <p>
                JavaScript/TypeScript
              </p>
            </div>
            <div
              class="skill"
            >
              <i
                class="icon fab fa-sass"
              />
              <p>
                Sass
              </p>
            </div>
            <div
              class="skill"
            >
              <i
                class="icon fab fa-python"
              />
              <p>
                Python
              </p>
            </div>
            <div
              class="skill"
            >
              <i
                class="icon fab fa-git"
              />
              <p>
                git
              </p>
            </div>
            <div
              class="skill"
            >
              <i
                class="icon fas fa-database"
              />
              <p>
                Databases
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      id="skills"
    >
      <div
        class="skills-card"
      >
        <h1>
          skills
        </h1>
        <div
          class="skills"
        >
          <div
            class="skill"
          >
            <i
              class="icon fab fa-react"
            />
            <p>
              React
            </p>
          </div>
          <div
            class="skill"
          >
            <i
              class="icon fab fa-node"
            />
            <p>
              Node/Express
            </p>
          </div>
          <div
            class="skill"
          >
            <i
              class="icon fab fa-html5"
            />
            <p>
              HTML
            </p>
          </div>
          <div
            class="skill"
          >
            <i
              class="icon fab fa-css3-alt"
            />
            <p>
              CSS
            </p>
          </div>
          <div
            class="skill"
          >
            <i
              class="icon fab fa-js-square"
            />
            <p>
              JavaScript/TypeScript
            </p>
          </div>
          <div
            class="skill"
          >
            <i
              class="icon fab fa-sass"
            />
            <p>
              Sass
            </p>
          </div>
          <div
            class="skill"
          >
            <i
              class="icon fab fa-python"
            />
            <p>
              Python
            </p>
          </div>
          <div
            class="skill"
          >
            <i
              class="icon fab fa-git"
            />
            <p>
              git
            </p>
          </div>
          <div
            class="skill"
          >
            <i
              class="icon fas fa-database"
            />
            <p>
              Databases
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
